appvia/kore
# Cost Estimation UI for Clusters

## Summary

Implements UI for cluster cost estimation, working for both AWS and GCP.

**Which issue(s) this PR resolves**:
Resolves #961 

## Checklist

 - [x] I've created a PR to update the [documentation](https://github.com/appvia/kore-docs): appvia/kore-docs#18

## Changes

* UI cost estimation / display for clusters. ef343eed
  * Use the metadata from the API to enrich UI with region drop-downs and machine-type drop-downs for both GCP and AWS
  * Calculate 'on the fly' estimates for node pools as the user edits
  * Request full plan estimates from the API when creating, editing and using plans.
* Ensure API estimates costs of non-autoscale clusters. d903398
* Sort schemas for GKE and EKS logically instead of alphabetically to improve UI display ordering. d32a8a2

### Additional changes

* Use a patched cloudinfo to work around issue retrieving AWS information in some regions. 5d1811c

## Screenshots

Cost estimate on plan create/edit:

![image](https://user-images.githubusercontent.com/7505593/87079080-74583780-c21d-11ea-9b5f-611077dffc12.png)

Cost estimate on node pool editor:

![image](https://user-images.githubusercontent.com/7505593/87079131-84701700-c21d-11ea-8d52-39e0b3da4afe.png)

## Examples

Enable costs support by applying `examples/kore-costs-feature.yaml` then play with the UI. You will need to ensure that the credential references in the kore-costs-feature point to valid AWS and GCP credentials.

## Testing

* Add new cluster plans for AWS and GCP
* Edit existing cluster plans for AWS and GCP
* Edit plan when creating a cluster for AWS and GCP and see the estimated costs being shown.

## Follow-up stories

* #1049 - use the new metadata API to power K8S version selection controls on the UI.
* #1050 - show cost estimates when viewing/editing existing clusters and viewing existing plans.
